### PR TITLE
Refactor and bug fixes around anonymous and temporary user states

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.m [hunk 1]
```diff
@@ -404,10 +404,7 @@ + (void)unlock:(BOOL)success action:(SFSecurityLockoutAction)action passcodeConf
 + (void)timerExpired:(NSTimer*)theTimer
 {
     [self log:SFLogLevelInfo msg:@"Inactivity NSTimer expired."];
-    [SFSecurityLockout setLockScreenFailureCallbackBlock:^{
-        [[SFAuthenticationManager sharedManager] logout];
-    }];
-	[SFSecurityLockout lock];
+    [SFSecurityLockout lock];
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.m]: This one is actually a duplicate callback post failure.

**Comment by bhariharan** [comment]
Thanks @jeanbovet for doing the heavy-lifting on this. I have tweaked things to fit in our current version, but it's pretty much the same as the work you did.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m [hunk 2]
```diff
@@ -584,45 +688,100 @@ - (SFUserAccount *)loadUserAccountFromFile:(NSString *)filePath {
...
+
+    // Try to load the user account assuming it is encrypted
+    SFUserAccount *user = nil;
+    if ([self loadUserAccountFromFile:filePath encrypted:YES account:&user error:nil]) {
+        return user;
+    } else {
+
+        // Unable to retrieve the encrypted user, maybe it's still in the old plain text format?
+        NSError *error = nil;
+        if ([self loadUserAccountFromFile:filePath encrypted:NO account:&user error:&error]) {
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m]: My ios9 fix (around the fact that unarchiveObjectWithFile doesn't throw an exception but return nil) need to make its way into loadUserAccountFromFile

Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m]: Good catch @wmathurin! I've added that check back in.

**Comment by jeanbovet** [comment]
Looks good, thanks @bhariharan !

**Comment by wmathurin** [comment]
:+1:

**Pull request merged**